src/domain/payments/payment-flow.ts

```diff
@@ -133,15 +159,49 @@ export const PaymentFlow = <S extends WalletCurrency, R extends WalletCurrency>(
 
   return {
     ...state,
-    protocolFeeInSenderWalletCurrency,
-    paymentAmounts,
-    totalAmountsForPayment,
-    routeDetails,
-    recipientDetails,
-    senderWalletDescriptor,
-    recipientWalletDescriptor,
-    checkBalanceForSend,
+    ...PaymentFlowBase(baseState),
```

Comment:
I don't like the word `Base` it implies some kind of OO hierarchy. What is really happening here? Code sharing? Polymorphism? Derivation?

Reply:
`Shared` would be more accurate.

The "Shared" object exposes  common methods and then:
- LightningFlow brings 2 additional unique methods
- OnChainFlow brings 3 additional unique methods

In this case it was useful to unify the implementation for those 8 common methods, but I can split that if we want to stick to the 3-minimum rule.